applications/sdp/mspi/src/main.c

```diff
@@ -52,215 +57,494 @@ typedef struct __packed {
-		nrf_vpr_clic_int_pending_set(NRF_VPRCLIC, VEVIF_IRQN(HRT_VEVIF_IDX_WRITE_SINGLE));
-	} else {
-		nrf_vpr_clic_int_pending_set(NRF_VPRCLIC, VEVIF_IRQN(HRT_VEVIF_IDX_WRITE_QUAD));
-	}
+	ll_prepare_transfer(xfer_modes.data, nrfe_mspi_xfer_packet.num_bytes * BITS_IN_BYTE);
+
+	nrf_vpr_clic_int_pending_set(NRF_VPRCLIC, VEVIF_IRQN(HRT_VEVIF_IDX_WRITE));
 }
 
-__attribute__((interrupt)) void hrt_handler_write_single(void)
+static void process_packet(const void *data, size_t len)
 {
-	xfer_ll_params.ce_enable_state = (mspi_dev_configs.ce_polarity == MSPI_CE_ACTIVE_HIGH);
+	(void)len;
+	uint16_t dir;
+	uint16_t out;
+	nrfe_mspi_flpr_response_t response;
+	uint8_t opcode = *(uint8_t *)data;
 
-	write_single_by_word(xfer_ll_params);
-}
+	switch (opcode) {
+	case NRFE_MSPI_CONFIG_PINS: {
+		nrfe_mspi_pinctrl_soc_pin_t *pins_cfg = (nrfe_mspi_pinctrl_soc_pin_t *)data;
 
-__attribute__((interrupt)) void hrt_handler_write_quad(void)
-{
-	xfer_ll_params.ce_enable_state = (mspi_dev_configs.ce_polarity == MSPI_CE_ACTIVE_HIGH);
+		response.opcode = pins_cfg->opcode;
+
+		ce_vios_count = 0;
+		data_vios_count = 0;
+		dir = 0;
+		out = 0;
+
+		for (uint8_t i = 0; i < NRFE_MSPI_PINS_MAX; i++) {
+			uint32_t psel = NRF_GET_PIN(pins_cfg->pin[i]);
+			uint32_t fun = NRF_GET_FUN(pins_cfg->pin[i]);
+
+			uint8_t pin_number = psel & PINCTL_PIN_NUM_MASK;
+
+			if (pin_number >= sizeof(pin_to_vio_map)) {
+				/* TODO: Jira ticket: NRFX-6875 error*/
+				return;
+			}
+
+			dir = BIT_SET_VALUE(dir, pin_to_vio_map[pin_number],
+					    VPRCSR_NORDIC_DIR_OUTPUT);
+
+			if ((fun >= NRF_FUN_SDP_MSPI_CS0) && (fun <= NRF_FUN_SDP_MSPI_CS4)) {
+
+				NRFX_ASSERT(pin_number < sizeof(pin_to_vio_map))
+				ce_vios[ce_vios_count] = pin_to_vio_map[pin_number];
+				ce_vios_count++;
+
+				/* TODO: Jira ticket: NRFX-6876 Get CE disabled states and set them,
+				 * they need to be passed from app
+				 */
+			} else if ((fun >= NRF_FUN_SDP_MSPI_DQ0) && (fun <= NRF_FUN_SDP_MSPI_DQ7)) {
+
+				NRFX_ASSERT(pin_number < sizeof(pin_to_vio_map))
+				data_vios[data_vios_count] = pin_to_vio_map[pin_number];
+				data_vios_count++;
+			}
+		}
+		nrf_vpr_csr_vio_dir_set(dir);
+		nrf_vpr_csr_vio_out_set(0);
+		break;
+	}
+	case NRFE_MSPI_CONFIG_CTRL: {
+		nrfe_mspi_cfg_t *cfg = (nrfe_mspi_cfg_t *)data;
+
+		response.opcode = cfg->opcode;
+		nrfe_mspi_cfg = cfg->cfg;
+		break;
+	}
+	case NRFE_MSPI_CONFIG_DEV: {
+		nrfe_mspi_dev_cfg_t *cfg = (nrfe_mspi_dev_cfg_t *)data;
+		response.opcode = cfg->opcode;
+		nrfe_mspi_dev_cfg = cfg->cfg;
 
-	write_quad_by_word(xfer_ll_params);
+		configure_clock(nrfe_mspi_dev_cfg.cpp);
+		break;
+	}
+	case NRFE_MSPI_CONFIG_XFER: {
+		nrfe_mspi_xfer_t *xfer = (nrfe_mspi_xfer_t *)data;
+
+		response.opcode = xfer->opcode;
+		nrfe_mspi_xfer = xfer->xfer;
+		break;
+	}
+	case NRFE_MSPI_TX:
+	case NRFE_MSPI_TXRX: {
+		nrfe_mspi_xfer_packet_t *packet = (nrfe_mspi_xfer_packet_t *)data;
+
+		response.opcode = packet->opcode;
+		nrfe_mspi_xfer_packet = packet->packet;
```

Comment:
Why is this saved in a global variable? If I checked correctly, only `prepare_and_send_data` and `ll_prepare_transfer` use it, the first one is called below, and the latter is called only in `prepare_and_send_data`. So, `mspi_xfer_packet` could just be passed as an argument to `prepare_and_send_data` and the global variable is not needed then.